src/psyclone/tests/psyGen_test.py

```diff
@@ -2617,8 +2627,269 @@ def test_binaryoperation_can_be_printed():
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("arg2").name == "arg2"
+    assert fake_parent.symbol_table.lookup("arg2").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("arg2").dimensions == 0
+    assert fake_parent.symbol_table.lookup("arg2").kind == 'read_arg'
+
+    reader = FortranStringReader("integer, intent( Out ) :: arg3")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("arg3").name == "arg3"
+    assert fake_parent.symbol_table.lookup("arg3").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("arg3").dimensions == 0
+    assert fake_parent.symbol_table.lookup("arg3").kind == 'write_arg'
+
+    reader = FortranStringReader("integer, intent ( InOut ) :: arg4")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("arg4").name == "arg4"
+    assert fake_parent.symbol_table.lookup("arg4").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("arg4").dimensions == 0
+    assert fake_parent.symbol_table.lookup("arg4").kind == 'rw_arg'
+
+    # Test different array dimensions
+    reader = FortranStringReader("integer, dimension(:,:,:) :: array1")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("array1").name == "array1"
+    assert fake_parent.symbol_table.lookup("array1").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("array1").dimensions == 3
+    assert fake_parent.symbol_table.lookup("array1").kind == 'local'
+
+    reader = FortranStringReader("real, dimension(:), intent(in) :: array2")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("array2").name == "array2"
+    assert fake_parent.symbol_table.lookup("array2").datatype == 'real'
+    assert fake_parent.symbol_table.lookup("array2").dimensions == 1
+    assert fake_parent.symbol_table.lookup("array2").kind == 'read_arg'
+
+    reader = FortranStringReader("real, intent(in), dimension(:) :: array3")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("array3").name == "array3"
+    assert fake_parent.symbol_table.lookup("array3").datatype == 'real'
+    assert fake_parent.symbol_table.lookup("array3").dimensions == 1
+    assert fake_parent.symbol_table.lookup("array3").kind == 'read_arg'
+
+    reader = FortranStringReader("integer, dimension(3,5) :: array4")
```

Comment:
In this case do we lose the information that the array is (3,5)? I thought that the `dimensions` component of the symbol would be `[3, 5]` in this case.

Reply:
Now it is. (in the 'shape' attribute)

Reply:
Great.